operators/src/adapters/raster_overlap_adapter.rs

```diff
@@ -282,21 +325,200 @@ pub fn insert_projected_pixels<'a, T: Pixel, I: Iterator<Item = &'a (GridIdx2D,
     Ok(())
 }
 
+/// This trait defines the behavior of the `RasterOverlapAdapter`.
+pub trait TileSubQuery<T>: Send
+where
+    T: Pixel,
+{
+    type FoldFuture: TryFuture<Ok = (RasterTile2D<T>, Self::FoldCompanion), Error = error::Error>;
+    type FoldMethod: Clone
+        + Fn((RasterTile2D<T>, Self::FoldCompanion), RasterTile2D<T>) -> Self::FoldFuture;
+    type FoldCompanion: Clone + Send;
+
+    /// The no-data-value to use in the resulting `RasterTile2D`
+    fn result_no_data_value(&self) -> Option<T>;
+    /// The initial fill-value of the accumulator (`RasterTile2D`).
+    fn initial_fill_value(&self) -> T;
+
+    /// This method geneerates a new accumulator which is used to fold the `Stream` of `RasterTile2D` of a sub-query.
+    fn new_fold_accu(
+        &self,
+        tile_info: TileInformation,
+        query_rect: QueryRectangle,
+    ) -> Result<(RasterTile2D<T>, Self::FoldCompanion)>;
+
+    /// This method generates a `QueryRectangle` for a tile-specific sub-query
+    fn tile_query_rectangle(
+        &self,
+        tile_info: TileInformation,
+        query_rect: QueryRectangle,
+        start_time: TimeInstance,
+    ) -> Result<QueryRectangle>;
+
+    /// This method generates the method which combines the accumulator and each tile of the sub-query stream in the `TryFold` stream adapter.
+    fn fold_method(&self) -> Self::FoldMethod;
+}
+
+#[derive(Debug, Clone)]
+pub struct TileSubQueryIdentity<F> {
+    fold_fn: F,
+}
+
+impl<T, FoldM, FoldF> TileSubQuery<T> for TileSubQueryIdentity<FoldM>
+where
+    T: Pixel,
+    FoldM: Send + Clone + Fn((RasterTile2D<T>, ()), RasterTile2D<T>) -> FoldF,
+    FoldF: TryFuture<Ok = (RasterTile2D<T>, ()), Error = error::Error>,
+{
+    fn result_no_data_value(&self) -> Option<T> {
+        Some(T::from_(0))
+    }
+
+    fn initial_fill_value(&self) -> T {
+        T::from_(0)
```

Comment:
Maybe even fill it with NO DATA if it is there :thinking:

Reply:
we don't know the NO DATA VALUE here

Reply:
cf. other comment

Reply:
there is a new task to add NO DATA VALUEE to RasterResultDescriptor